platform/windows/display_server_windows.cpp

```diff
@@ -3879,6 +3879,7 @@ void DisplayServerWindows::swap_buffers() {
 
 void DisplayServerWindows::set_native_icon(const String &p_filename) {
 	_THREAD_SAFE_METHOD_
+	MutexLock lock(icon_mutex);
```

Comment:
This is a thread safe class, the previous line will already lock the built-in mutex, so no additional interlocking is needed.